lib/improver/tests/spotdata/spotdata/test_framework.py

```diff
@@ -0,0 +1,278 @@
+                                            ('dz', 'f8'),
+                                            ('edgepoint', 'bool_')])
+
+        self.ancillary_directory = mkdtemp()
+        self.data_directory = mkdtemp()
+        self.configs = mkdtemp()
+        self.output_directory = mkdtemp()
+
+        data_path = self.data_directory + '/temperature_at_screen_level.nc'
+        orography_path = self.ancillary_directory + '/orography.nc'
+        land_path = self.ancillary_directory + '/land_mask.nc'
+        ad_path_temperature = (self.data_directory +
+                               '/temperature_on_height_levels.nc')
+        ad_path_pressure = (self.data_directory +
+                            '/pressure_on_height_levels.nc')
+        ad_path_s_pressure = self.data_directory + '/surface_pressure.nc'
+
+        iris.save(cube, data_path)
+        iris.save(orography, orography_path)
+        iris.save(land, land_path)
+        iris.save(temperature_on_height_levels, ad_path_temperature)
+        iris.save(pressure_on_height_levels, ad_path_pressure)
+        iris.save(surface_pressure, ad_path_s_pressure)
+
+        self.config_path = self.configs + '/spotdata_constants.json'
+        ff = open(self.config_path, 'w')
+        json.dump(diagnostic_recipe, ff, sort_keys=True, indent=4,
+                  separators=(',', ': ',))
+        ff.close()
+
+        self.made_files = [data_path, orography_path, land_path,
+                           ad_path_temperature, ad_path_pressure,
+                           ad_path_s_pressure, self.config_path]
+
+        self.cube = cube
+        self.ancillary_data = ancillary_data
+        self.additional_data = additional_data
+        self.sites = sites
+        self.neighbour_list = neighbour_list
+        self.diagnostic_recipe = diagnostic_recipe
+
+        self.args = (self.config_path,
+                     self.data_directory,
+                     self.ancillary_directory)
+
+        site_properties = {
+            'latitudes': [10, 20, 30, 40, 50, 60],
+            'longitudes': [0, 0, 0, 0, 0, 0],
+            'altitudes': [0, 1, 2, 3, 4, 5],
+            'site_ids': [0, 1, 2, 3, 4, 5],
+            }
+
+        self.kwargs = {
+            'diagnostic_list': ['temperature'],
+            'site_path': None,
+            'constants_file_path': None,
+            'site_properties': site_properties,
+            'forecast_date': '20170217',
+            'forecast_time': 6,
+            'forecast_length': 2,
+            'output_path': self.output_directory,
+            'use_multiprocessing': False
+            }
+
+    def tearDown(self):
+        """Remove temporary directories created for testing."""
+        for a_file in self.made_files:
+            Call(['rm', a_file])
+        Call(['rmdir', self.ancillary_directory])
+        Call(['rmdir', self.data_directory])
+        Call(['rmdir', self.configs])
+        Call(['rmdir', self.output_directory])
+
+
```

Comment:
Agreed. I've simplified to be only the necessary stuff to cover the three tests. I chose a simpler method of extracting data, so that removed the need for all the multi-level data.